src/services/context/context.factory.ts

```diff
@@ -44,28 +41,49 @@ export class WizdomContextFactory {
                 return result.json().then((json) => {
                     if (json.wizdom_x002e_properties) {
                         var context: IWizdomContext = JSON.parse(json.wizdom_x002e_properties);
-                        // ensure tailing / for app- and bloburl
-                        if (context.blobUrl.substr(-1) != "/")
-                            context.blobUrl = context.blobUrl + "/";
-                        if (context.appUrl.substr(-1) != "/")
-                            context.appUrl = context.appUrl + "/";
-
                         this.allPropertiesContext = context;
                     }
                 });
             });
 
-            return Promise.all([storageEntityPromise, allPropertiesPromise]).then(() => {
-                if (this.allPropertiesContext) {
-                    if (this.allPropertiesContext.appUrl)
-                        this.storageEntityContext.appUrl = this.allPropertiesContext.appUrl
-                    if (this.allPropertiesContext.blobUrl)
-                        this.storageEntityContext.blobUrl = this.allPropertiesContext.blobUrl
-                    if (this.allPropertiesContext.clientId)
-                        this.storageEntityContext.clientId = this.allPropertiesContext.clientId
+            await Promise.all([storageEntityPromise, allPropertiesPromise])
+            
+            if (this.allPropertiesContext) {
+                if (this.allPropertiesContext.appUrl)
+                    this.storageEntityContext.appUrl = this.allPropertiesContext.appUrl
+                if (this.allPropertiesContext.blobUrl)
+                    this.storageEntityContext.blobUrl = this.allPropertiesContext.blobUrl
+                if (this.allPropertiesContext.clientId)
+                    this.storageEntityContext.clientId = this.allPropertiesContext.clientId
+                this.storageEntityContext.isWizdomSaaS = this.allPropertiesContext.isWizdomSaaS ?? this.storageEntityContext.isWizdomSaaS;
+            }
+
+            if(this.storageEntityContext.isWizdomSaaS || this.wizdomdevelopermode?.wizdomContext?.isWizdomSaaS) {
+
+                // Make sure to use the appUrl from wizdomdevelopermode if it's there, otherwise default down the chain till we find something
+                // Use local variable to avoid caching the developermode appUrl
+                let appUrl = this.wizdomdevelopermode?.wizdomContext?.appUrl ?? (this.storageEntityContext.appUrl || "https://wtsaas.azurewebsites.net/")
+
+                let httpClient: AadHttpClient = await this.aadHttpClientPromise;
+                let wizdomInfo: IWizdomContext = await (await httpClient.fetch(appUrl + "api/wizdom/365/context", AadHttpClient.configurations.v1, {method: "GET"})).json()
```

Comment:
We've already asked tenant properties for context. Maybe optimize by only calling this, if we dont have all properties already in tenant props.

Reply:
This is the stuff we normally get from corsProxy and should NOT be removed. It also returns general information usually stored in tenant properties in an effort to get rid of or reduce the need for tenant properties as much as possible

Reply:
Yeah I later found out, and updated the comment (O; some of these are just notes, as David is also going thou the PR.